src/Engine/Component/CollisionComponents/BounderComponent.cpp

```diff
@@ -0,0 +1,152 @@
+#include "BounderComponent.hpp"
+
+#include <tuple>
+
+#include "glm/gtx/component_wise.hpp"
+#include "glm/gtx/norm.hpp"
+
+#include "Component/SpatialComponents/SpatialComponent.hpp"
+
+
+
+BounderComponent::BounderComponent(unsigned int weight) :
+    m_spatial(nullptr),
+    m_weight(weight)
+{}
+
+void BounderComponent::init(GameObject & go) {
+    Component::init(go);
+    if (!(m_spatial = gameObject()->getSpatial())) assert(false);
+}
+
+
+
+AABBounderComponent::AABBounderComponent(unsigned int weight, const AABox & box) :
+    BounderComponent(weight),
+    m_box(box),
+    m_transBox(m_box)
+{}
+
+void AABBounderComponent::update(float dt) {
+    // no rotation
+    if (m_spatial->orientationMatrix() == glm::mat3()) {
+        m_transBox.min = m_box.min * m_spatial->scale() + m_spatial->position();
+        m_transBox.max = m_box.max * m_spatial->scale() + m_spatial->position();
+    }
+    // is rotation
+    else {
+        glm::vec3 corners[8]{
+            glm::vec3(m_box.min.x, m_box.min.y, m_box.min.z),
+            glm::vec3(m_box.max.x, m_box.min.y, m_box.min.z),
+            glm::vec3(m_box.min.x, m_box.max.y, m_box.min.z),
+            glm::vec3(m_box.max.x, m_box.max.y, m_box.min.z),
+            glm::vec3(m_box.min.x, m_box.min.y, m_box.max.z),
+            glm::vec3(m_box.max.x, m_box.min.y, m_box.max.z),
+            glm::vec3(m_box.min.x, m_box.max.y, m_box.max.z),
+            glm::vec3(m_box.max.x, m_box.max.y, m_box.max.z)
+        };
+        const glm::mat4 & modelMat(m_spatial->modelMatrix());
+        for (int i(0); i < 8; ++i) {
+            corners[i] = modelMat * glm::vec4(corners[i], 1.0f);
+        }
+        m_transBox.min = corners[0];
+        m_transBox.max = corners[0];
+        for (int i(1); i < 8; ++i) {
+            m_transBox.min = glm::min(m_transBox.min, corners[i]);
+            m_transBox.max = glm::max(m_transBox.max, corners[i]);
+        }
+    }
+}
+
+bool AABBounderComponent::collide(const BounderComponent & o, glm::vec3 * delta) const {
+    if (dynamic_cast<const AABBounderComponent *>(&o)) 
```

Comment:
Does dynamic_cast have a lot of overhead? Would it be worth replacing all the dynamic_casts throughout the engine with enums and templates?